packages/react-router/modules/matchPath.js

```diff
@@ -26,11 +27,13 @@ const compilePath = (pattern, options) => {
 /**
  * Public API for matching a URL pathname to a path pattern.
  */
-const matchPath = (pathname, path, options = {}) => {
+const matchPath = (pathname, path, options = {}, parent = null) => {
   const { exact = false, strict = false } = options
 
   if (!path)
-    return { url: pathname, isExact: true, params: {} }
+    return { url: parent ? parent.url : '/', isExact: true, params: {} }
```

Comment:
I had to change the `url` from `pathname` to whatever its parent `match`'s `url` is because it otherwise was breaking relative matching. The full location is still available through the `location` prop. If someone is using a pathless `<Route>`, I'm not even sure why they would be checking the `match`.